pkg/ebpf/http.go

```diff
@@ -0,0 +1,196 @@
+// +build linux_bpf
+
+package ebpf
+
+import (
+	"fmt"
+	"time"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/pkg/ebpf/bytecode"
+	"github.com/DataDog/datadog-agent/pkg/network"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/ebpf"
+	"github.com/DataDog/ebpf/manager"
+)
+
+/*
+#include "c/tracer-ebpf.h"
+*/
+import "C"
+
+const (
+	HTTPBatchSize  = int(C.HTTP_BATCH_SIZE)
+	HTTPBatchPages = int(C.HTTP_BATCH_PAGES)
+)
+
+type httpTX C.http_transaction_t
+type httpNotification C.http_batch_notification_t
+type httpBatch C.http_batch_t
+
+func toHTTPNotification(data []byte) httpNotification {
+	return *(*httpNotification)(unsafe.Pointer(&data[0]))
+}
+
+// Path returns the URL from the request fragment captured in eBPF
+// Usually the request fragment will look like
+// GET /foo HTTP/1.1
+func (tx *httpTX) Path() string {
+	b := C.GoBytes(unsafe.Pointer(&tx.request_fragment), C.int(C.HTTP_BUFFER_SIZE))
```

Comment:
Since it seems you concerned about performance of this function (due to the loops below, rather than something like `strings.Split`), if you aren't aware,  `C.GoBytes` will copy (and thus allocate). If you want to iterate on the raw bytes, you can instead do `(*[C.HTTP_BUFFER_SIZE]byte)(unsafe.Pointer(&tx.request_fragment))` and then only do a `copy` once you've found the range you are interested in.

You may also want to cache that return value with a `sync.Once` or something, if this function may be called a lot.